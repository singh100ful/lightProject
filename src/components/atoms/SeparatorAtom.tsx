import React from 'react';
import {View, StyleSheet} from 'react-native';
import {scalePresets} from '../../theme/scale';
import {ColorPresets} from '../../theme/color';

interface SeparatorAtomProps {
  height?: number;
}

export const SeparatorAtom: React.FC<SeparatorAtomProps> = ({height}) => {
  return (
    <View style={styles.container}>
      <View style={[styles.line, {height: height ? height : 1}]} />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-around',
    paddingVertical: scalePresets.mediumScale,
    paddingHorizontal: scalePresets.baseScale,
  },
  line: {flex: 1, backgroundColor: ColorPresets.border},
});
