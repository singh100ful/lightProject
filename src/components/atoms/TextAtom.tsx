import * as React from 'react';
import {Text, TextProps, TextStyle} from 'react-native';
import {ColorPresets} from '../../theme/color';
import {defaultTexts} from '../../theme/fonts';

interface TextAtomProps extends TextProps {
  text: string;
  style?: TextStyle;
  preset?: TextPresets;
}

export const TextAtom: React.FC<TextAtomProps> = ({
  text,
  style,
  preset = 'body',
  ...rest
}) => {
  const textStyle = [Presets[preset], style];
  return (
    <Text
      style={[
        {
          flexShrink: 1,
          flexWrap: 'wrap',
          color: ColorPresets.primaryText,
        },
        textStyle,
      ]}
      {...rest}>
      {text}
    </Text>
  );
};

const Presets = {
  title1: {...defaultTexts.title1} as TextStyle,
  title2: {...defaultTexts.title2} as TextStyle,
  bodyBold: {...defaultTexts.bodyBold} as TextStyle,
  body: {...defaultTexts.body} as TextStyle,
  caption: {...defaultTexts.caption} as TextStyle,
};

export type TextPresets = keyof typeof Presets;
