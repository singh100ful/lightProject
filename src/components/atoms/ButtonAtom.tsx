import * as React from 'react';
import {Pressable, PressableProps, StyleSheet} from 'react-native';
import {TextAtom} from './TextAtom';
import {scalePresets} from '../../theme/scale';
import {ColorPresets} from '../../theme/color';

interface ButtonAtomProps extends PressableProps {
  title: string;
}

export const ButtonAtom: React.FC<ButtonAtomProps> = ({title, ...rest}) => {
  return (
    <Pressable style={styles.container} {...rest}>
      <TextAtom text={title} preset="title2" />
    </Pressable>
  );
};

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: scalePresets.baseScale,
    paddingVertical: scalePresets.mediumScale,
    alignItems: 'center',
    backgroundColor: ColorPresets.primaryRed,
    marginVertical: scalePresets.mediumScale,
    borderRadius: scalePresets.mediumScale,
  },
});
