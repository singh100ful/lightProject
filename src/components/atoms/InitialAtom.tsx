import * as React from 'react';
import {View, StyleSheet} from 'react-native';
import {TextAtom} from './TextAtom';
import {moderateScale} from '../../theme/scale';
import {scaleFont} from '../../theme/fonts';

interface InitialAtomProps {
  initial: string;
}

export const InitialAtom: React.FC<InitialAtomProps> = props => {
  const [color, setColor] = React.useState('');

  const getRandomColor = () => {
    var letters = '0123456789ABCDEF';
    var color = '#';
    for (var i = 0; i < 6; i++) {
      color += letters[Math.floor(Math.random() * 16)];
    }
    setColor(color);
  };

  React.useEffect(() => {
    getRandomColor();
  }, []);

  return (
    <View style={[styles.container, {backgroundColor: color}]}>
      <TextAtom style={styles.initials} text={props.initial.charAt(0)} />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: moderateScale(100),
    width: moderateScale(50),
    height: moderateScale(50),
  },
  initials: {color: 'white', ...scaleFont(20)},
});
