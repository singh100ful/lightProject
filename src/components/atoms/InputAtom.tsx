import * as React from 'react';
import {StyleSheet, TextInput, TextInputProps, View} from 'react-native';
import {TextAtom} from './TextAtom';
import {ColorPresets} from '../../theme/color';
import {scalePresets} from '../../theme/scale';
import {defaultTexts} from '../../theme/fonts';

interface InputAtomProps extends TextInputProps {
  label: string;
}

export const InputAtom: React.FC<InputAtomProps> = ({label, ...rest}) => {
  const [focus, setfocus] = React.useState(false);
  return (
    <View style={styles.container}>
      {label ? (
        <View style={styles.labelStyle}>
          <TextAtom
            text={label}
            style={{
              color: focus ? ColorPresets.primaryColor : ColorPresets.border,
            }}
          />
        </View>
      ) : null}
      <TextInput
        {...rest}
        onFocus={() => setfocus(true)}
        onBlur={() => setfocus(false)}
        placeholderTextColor={ColorPresets.placeholderText}
        style={[
          styles.input,
          {
            borderColor: focus
              ? ColorPresets.primaryColor
              : ColorPresets.border,
          },
        ]}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {paddingVertical: scalePresets.baseScale},
  input: {
    ...defaultTexts.placeHolder,
    fontWeight: '400',
    backgroundColor: ColorPresets.white,
    borderRadius: scalePresets.smallScale,
    borderWidth: 1,
    paddingHorizontal: scalePresets.mediumScale,
    paddingVertical: scalePresets.mediumScale,
  },
  labelStyle: {
    position: 'absolute',
    left: scalePresets.mediumScale,
    top: scalePresets.smallScale,
    paddingHorizontal: scalePresets.smallScale,
    backgroundColor: ColorPresets.white,
    zIndex: 1,
  },
});
