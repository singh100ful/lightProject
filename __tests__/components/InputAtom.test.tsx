import React from 'react';
import renderer, {act} from 'react-test-renderer';
import {InputAtom} from '../../src/components/atoms/InputAtom';
import {Text, TextInput} from 'react-native';
import {TextAtom} from '../../src/components/atoms/TextAtom';

describe('Input Atom', () => {
  it('renders correctly with placeholder', () => {
    const tree = renderer
      .create(<InputAtom label="Label" placeholder="Enter text" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('displays the correct label', () => {
    const component = renderer.create(
      <InputAtom label="Username" placeholder="Enter your username" />,
    );
    const labelInstance = component.root.findByType(TextAtom);

    expect(labelInstance.props.text).toBe('Username');
  });

  it('captures and displays input correctly', () => {
    const component = renderer.create(
      <InputAtom label="Label" placeholder="Enter text" />,
    );
    const inputInstance = component.root.findByType(TextInput);

    act(() => {
      inputInstance.props.onChangeText('Hello World');
    });

    expect(inputInstance.props.value).toBe('Hello World');
  });
});
