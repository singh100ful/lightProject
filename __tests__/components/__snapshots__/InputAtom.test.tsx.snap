// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Atom renders correctly with placeholder 1`] = `
<View>
  <Text
    style={
      [
        {
          "color": "#000000",
          "flexShrink": 1,
          "flexWrap": "wrap",
        },
        undefined,
      ]
    }
  >
    Label
  </Text>
  <TextInput
    placeholder="Enter text"
    placeholderTextColor="#949494"
    style={
      {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 7.708333333333334,
        "borderWidth": 1,
        "paddingHorizontal": 15.416666666666668,
        "paddingVertical": 15.416666666666668,
      }
    }
  />
</View>
`;
